packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -81,36 +81,30 @@ export class LibraryEngine extends Engine<undefined> {
     version: string
   }
 
-  constructor(config: EngineConfig, loader: LibraryLoader = new DefaultLibraryLoader(config)) {
+  constructor(config: EngineConfig, libraryLoader?: LibraryLoader) {
     super()
 
-    try {
-      // we try to handle the case where the datamodel is not found
-      this.datamodel = fs.readFileSync(config.datamodelPath, 'utf-8')
-    } catch (e) {
-      if ((e.stack as string).match(/\/\.next|\/next@|\/next\//)) {
-        throw new PrismaClientInitializationError(
-          `Your schema.prisma could not be found, and we detected that you are using Next.js.
-Find out why and learn how to fix this: https://pris.ly/d/schema-not-found-nextjs`,
-          config.clientVersion!,
-        )
-      } else if (config.isBundled === true) {
-        throw new PrismaClientInitializationError(
-          `Prisma Client could not find its \`schema.prisma\`. This is likely caused by a bundling step, which leads to \`schema.prisma\` not being copied near the resulting bundle. We would appreciate if you could take the time to share some information with us.
-Please help us by answering a few questions: https://pris.ly/bundler-investigation-error`,
-          config.clientVersion!,
-        )
-      }
+    const engineType = getClientEngineType(config.generator!)
 
-      throw e
+    if (TARGET_BUILD_TYPE === 'library') {
+      // for "library" builds, we can use both the wasm and native engines
+      if (engineType === ClientEngineType.Wasm && config.adapter !== undefined) {
+        this.libraryLoader = libraryLoader ?? wasmLibraryLoader
+      } else {
+        this.libraryLoader = libraryLoader ?? defaultLibraryLoader
+      }
+    } else {
+      // any other build using LibraryEngine, only wasm engine can be used
+      this.libraryLoader = libraryLoader ?? wasmLibraryLoader
     }
 
     this.config = config
     this.libraryStarted = false
     this.logQueries = config.logQueries ?? false
     this.logLevel = config.logLevel ?? 'error'
-    this.libraryLoader = loader
     this.logEmitter = config.logEmitter
+    this.datamodel = atob(config.inlineSchema)
```

Comment:
Not necessary something for now, but we should consider removing base64-encoding for inline schema here - where for DataProxy it could be argued that we are saving some time on encoding it (although, I'd say DP itself has no reason to require schema to be base64-encoded), in case of NodeAPI/WASM it does nothing but increase generated bundle size by 33% and require us to spend some time on decoding during startup.

Reply:
I agree. We went base64 because the binary engine gets started with the schema that way. I think we could introduce a new cli option to start it with the schema as is. Do you know if there was any reason for the base64?

Reply:
Possibly relevant: How do we upload the schema to DataProxy/Accelerate?

Reply:
https://prisma-company.slack.com/archives/C04TW4A2H8C/p1698863742436789?thread_ts=1698787796.270229&cid=C04TW4A2H8C

Reply:
This is also now tracked in an issue